File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        prefix = [0]
        for i in range(0, len(nums)):
            prefix.append(prefix[i]+nums[i])

        res = float('inf')
        for i in range(len(nums)):
            l, r = 0, i+1
            while l <= r:
                m = (l+r)//2
                if prefix[i+1] - prefix[m] >= target:
                    res = min(res, i-m+1)
                    l = m + 1
                else:
                    r = m - 1

        return res if res < float('inf') else 0

File: Data Structures & Algorithms/search-insert-position/submission-2.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        let mut l: usize = 0;
        let mut r: usize = nums.len()-1;

        if target > nums[r] {
            return (r + 1) as i32
        }

        while l < r {
            let m = (l + r) / 2;
            if nums[m] > target {
                r = m;
            } else if nums[m] < target {
                l = m + 1;
            } else {
               return m as i32;
            }
        }

        l as i32
    }
}
